### ISSUE TITLE
Not processing exclude_archetype and extend_archetype YAML

### ISSUE THREAD

**Issue by jleonelion** [Azure/terraform-azurerm-caf-enterprise-scale#278]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 1.0.11

**azure provider**: >= 2.77.0

**module**: archetypes

### Description

Cannot override built in archetypes using yaml format.

Module is attempting to decode yaml files using `jsondecode` function instead of `yamldecode` function.  Since file content is yaml, jsondecode is not happy.

Need to update function called in these lines:
https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/main/modules/archetypes/locals.archetype_definitions.tf#L56
https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/main/modules/archetypes/locals.archetype_definitions.tf#L48

**Steps to Reproduce**
- Setup TF project to as described in [Expand built in archetype definitions](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/main/docs/wiki/%5BExamples%5D-Expand-built-in-archetype-definitions.md)
- Create a YAML based, not json like in the example, extend_archetype definitions file to override behavior a built-in archetype (such as es_landing_zones)
- Run `terraform plan`
- Blows up b/c module attempting to parse using `jsondecode`
```
│ Error: Error in function call
│ 
│   on /home/vscode/.terraform.cache/md7/modules/enterprise_scale/modules/archetypes/locals.archetype_definitions.tf line 56, in locals:
│   56:     filepath => jsondecode(templatefile("${local.custom_library_path}/${filepath}", local.template_file_vars))
│     ├────────────────
│     │ local.custom_library_path is "/tf/caf/enterprise-scale/lib"
│     │ local.template_file_vars is object with 25 attributes
│ 
│ Call to function "jsondecode" failed: invalid character 'e' looking for beginning of value.
```

**Comment by jleonelion** [comment]
Work-around is to stick with json format, but that sticks b/c can't add comments to the JSON file so people understand rational for why the built-in archetypes are being overridden.

**Comment by krowlandson** [comment]
Great spot... thank you @jleonelion

I have added a quick PR to add this and will release as a patch once approved by the team (hopefully tomorrow).

**Comment by krowlandson** [comment]
- This has been included in [release v1.1.3](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/releases/tag/v1.1.3) so please test and confirm so we can close this issue.
